### PR TITLE
Cerebro deployment

### PR DESCRIPTION
**Pull request by gianrubio**
This PR will create deployment for cerebro (replacement for kopf) and simplify csr for kibana, node and cerebro.
There's no official imagem from the upstream, so I would recommend to create a new one

```FROM openjdk:8-jre-slim

RUN groupadd -r cerebro && \
    useradd -g cerebro cerebro

ENV CEREBRO_VERSION=0.6.8
ADD https://github.com/lmenezes/cerebro/releases/download/v${CEREBRO_VERSION}/cerebro-${CEREBRO_VERSION}.tgz /usr/local/

RUN mv /usr/local/cerebro-${CEREBRO_VERSION} /opt/cerebro && \
    chown -R cerebro:cerebro/opt/cerebro

WORKDIR /opt/cerebro
USER cerebro
CMD [ "/opt/cerebro/bin/cerebro" ]
```

### CODE REVIEW

File: pkg/k8sutil/deployments.go [hunk 1]
```diff
@@ -384,3 +389,105 @@ func (k *K8sutil) CreateKibanaDeployment(baseImage, clusterName, namespace strin
...
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							v1.Container{
+								Name:            deploymentName,
+								Image:           baseImage,
+								ImagePullPolicy: "Always",
+								Command: []string{
+									"bin/cerebro",
+									"Dconfig.file=/opt/cerebro/cfg/application.conf",
```
Comment (stevesloka) [inline pkg/k8sutil/deployments.go]: This needs a `-` added. Based on the docker file I built, this is what I was testing with: 

```
"/usr/local/cerebro/bin/cerebro",
"-Dconfig.file=/opt/cerebro/cfg/application.conf",
```

File: pkg/k8sutil/certs.go [hunk 2]
```diff
@@ -294,6 +270,8 @@ func (k *K8sutil) CreateCertsSecret(namespace, clusterName, certsDir string) err
 			"node-key.pem":      nodeKey,
 			"kibana-key.pem":    kibanaKey,
 			"kibana.pem":        kibana,
+			"cerebro-key.pem":   cerebroKey,
```
Comment (stevesloka) [inline pkg/k8sutil/certs.go]: Are these certs that are generated used? Would be nice to serve cerebro over tls via these certs.

File: pkg/k8sutil/k8sutil.go [hunk 3]
```diff
@@ -583,3 +584,43 @@ func (k *K8sutil) CreateDataNodeDeployment(deploymentType string, replicas *int3
...
+
+func (k *K8sutil) CreateCerebroConfiguration(clusterName, cert string) map[string]string {
+	clusterName = fmt.Sprintf("https://%s:9200", fmt.Sprintf(fmt.Sprintf("elasticsearch-%s", clusterName)))
+
+	x := map[string]string{}
+	x["application.conf"] = fmt.Sprintf(`
+play.ws.ssl {
+        trustManager = {
+                stores = [
+				{ type = "PEM", path = "%s/cerebro.pem" },
```
Comment (stevesloka) [inline pkg/k8sutil/k8sutil.go]: I think you're using the cert to talk to the ES cluster, please confirm, but makes sense.

**Comment by stevesloka** [comment]
I created the docker image for cerebro: `upmcenterprises/cerebro:0.6.8` (https://github.com/upmc-enterprises/cerebro-docker). 

It needs a bit of work to run as user `cerebro` as you had in your example, however.

File: pkg/k8sutil/k8sutil.go (hunk 3 above)
Comment (gianrubio) [reply pkg/k8sutil/k8sutil.go]: yes, that's the point

File: pkg/k8sutil/deployments.go (hunk 1 above)
Comment (gianrubio) [reply pkg/k8sutil/deployments.go]: Done

**Comment by gianrubio** [comment]
@stevesloka I tested my last commit in a new namespace and all the components successfully started

**Comment by stevesloka** [comment]
Looks good, thanks @gianrubio!

**Pull request merged**